### PR TITLE
Crud implementation

### PR DESCRIPTION
**Pull request by JasinskiyD**

### CODE REVIEW

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 1]
```diff
@@ -1,116 +1,135 @@
...
+    }
+
+    // checks the parameter that method accepts (parameter is the number of the element in the list), returns true if range acceptable and false if not
+    private boolean indexInRange(int index) {
+        return 0 <= index && index < size;
+    }
+
+    // add new element in the end of list, if element is not acceptable return - false, if element added return - true
+    public boolean add(String element) {
+        if (element == null) {  // if element is invalid
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:39]: please, move comment above code

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 2]
```diff
@@ -1,116 +1,135 @@
...
     private int size;
 
+    // return size of the list
     public int size() {
         return size;
     }
 
+    // check if list is empty and returns status (true if empty and false if not)
     public boolean isEmpty() {
         return size == 0;
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:14]: replace with following head == null

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 3]
```diff
@@ -1,116 +1,135 @@
...
+    private boolean indexInRange(int index) {
+        return 0 <= index && index < size;
+    }
+
+    // add new element in the end of list, if element is not acceptable return - false, if element added return - true
+    public boolean add(String element) {
+        if (element == null) {  // if element is invalid
+            return false;   // method return false
+        }
+        if (head == null) {  // if first element in list is null
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:42]: replace with isEmpty()

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 4]
```diff
@@ -1,116 +1,135 @@
...
-        size++;
+        size++; // update size of the list
         return true;
     }
 
+
     public String get(int index) {
-        int currentIndex = 0;
-        Node temp = head;
+        if (indexInRange(index)) {
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:53]: replace with negative operator

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 5]
```diff
@@ -1,116 +1,135 @@
...
-        if (index < 0 || index >= size) {
-            throw new IndexOutOfBoundsException();
-        }
-
-        if (index == 0) {
-            temp.setValue(value);
-            return;
-        }
+    public void set(int index, String element) {
+        if (indexInRange(index)) {
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:69]: see comment in row #53

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 6]
```diff
@@ -1,116 +1,135 @@
...
+                temp.setValue(element);
                 return;
-            } else {
-                temp = temp.getNext();
-                currentIndex++;
+            }
+
+            while (temp != null) {
+                if ((currentIndex + 1) == index) {
+                    currentValue = new Node(element);
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:81]: currentValue. May be should be newNode?

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 7]
```diff
@@ -1,116 +1,135 @@
...
-            throw new IndexOutOfBoundsException();
-        }
-        if (index == 0) {
-            head = head.getNext();
-            size--;
-            return;
-        }
+    public void remove(int index) {
+        if (indexInRange(index)) {
+            if (isEmpty()) {
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:96]: seems this is useless

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 8]
```diff
@@ -1,116 +1,135 @@
...
-        if (index < 0 || index >= size) {
-            throw new IndexOutOfBoundsException();
-        }
-        if (index == 0) {
-            head = head.getNext();
-            size--;
-            return;
-        }
+    public void remove(int index) {
+        if (indexInRange(index)) {
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:95]: revert condition

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 9]
```diff
@@ -1,116 +1,135 @@
...
+    // check if list is empty and returns status (true if empty and false if not)
     public boolean isEmpty() {
         return size == 0;
     }
 
-    public boolean add(String value) {
-        if (value == null) {
-            throw new NullPointerException("The first argument for add() is null.");
+    // returns last list element
+    private Node getLastNode() {
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:18]: refactor this tone o Node getNode(int index);

File: src/main/java/myLinkedList/MyLinkedList.java [hunk 10]
```diff
@@ -1,116 +1,135 @@
...
+                }
             }
         }
     }
 
+    // returns all elements of list in String format
     @Override
     public String toString() {
-        StringBuilder s = new StringBuilder();
+        StringBuilder string = new StringBuilder();
```
Comment (sergey-chumachenko) [inline src/main/java/myLinkedList/MyLinkedList.java:124]: please, replace this with StrBuilder form apache commons library

**Review by sergey-chumachenko — CHANGES_REQUESTED** [review]

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 1 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:39]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 2 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:14]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 3 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:42]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 5 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:69]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 6 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:81]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 7 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:96]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 8 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:95]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 4 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:53]: done

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 10 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:124]: StrBuilder is depracated
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/text/StrBuilder.html

File: src/main/java/myLinkedList/MyLinkedList.java (hunk 9 above)
Comment (JasinskiyD) [reply src/main/java/myLinkedList/MyLinkedList.java:18]: done

**Pull request merged**